src/operations/shrink_wrap.rs

```diff
@@ -0,0 +1,80 @@
+use std::slice;
+use std::u32;
+
+use crate::convert::{cast_u32, cast_usize};
+
+use crate::geometry::{self, Geometry};
+
+pub struct ShrinkWrapParams {
+    pub geometry: Geometry,
+    pub sphere_density: u32,
+    pub step: u32,
+}
+
+struct ShrinkWrapState {
+    sphere_geometry: Geometry,
+    current_step: u32,
+}
+
+pub struct ShrinkWrapOp {
+    params: ShrinkWrapParams,
+    state: Option<ShrinkWrapState>,
+}
+
+impl ShrinkWrapOp {
+    pub fn new(params: ShrinkWrapParams) -> Self {
+        Self {
+            params,
+            state: None,
+        }
+    }
+
+    pub fn next_value(&mut self) -> Option<Geometry> {
+        if self.state.is_none() {
+            self.init();
+        }
```

Comment:
A very good question! It will be what we decide it is, but my reasoning for doing it this way is that the interpreter might want to spawn the operation but not yet make it do any work (at least I believe this will be a real use-case  currently). That means creating an operation should be cheap, but this operation does a lot of work in the `init()` method. Thoughts?

Reply:
I cannot imagine doing this for now without more info and examples. But it's not a biggie, it can stay like this, maybe with some extra comment to prevent confusion in the future.